lumicks/pylake/scan.py

```diff
@@ -36,7 +36,22 @@ def __repr__(self):
         return f"{name}(pixels=({self.pixels_per_line}, {self.lines_per_frame}))"
 
     def __getitem__(self, item):
-        raise NotImplementedError("Indexing and slicing are not implemented for scans")
+        """All indexing is in frames"""
+        ts_ranges = self.frame_timestamp_ranges(exclude=False)
+
+        if isinstance(item, slice):
+            if item.step is not None:
+                raise IndexError("Slice steps are not supported")
+            start = ts_ranges[0 if item.start is None else item.start][0]
+            stop = ts_ranges[self.num_frames - 1 if item.stop is None else item.stop - 1][-1]
```

Comment:
This logic looks a bit complicated and isn't really necessary?

It also misses a few corner cases. For example:
`scan[-more_than_the_frame_count:]` or `scan[:more_than_the_frame_count]`

Could just let Python handle it?
```python
sliced_ts_ranges = ts_ranges[item]
if not sliced_ts_ranges:
  raise NotImplementedError("Slice is empty.")

start, stop = sliced_ts_ranges[0][0], sliced_ts_ranges[-1][-1]
```

Quickly trying it passes all tests, but maybe I'm overlooking something?

Reply:
good catch! Implemented this now and took advantage of the new copy functionality for confocal classes. Also added some additional assertions to the tests to verify `num_frames` is correct